x/observer/migrations/v11/migrate_test.go

```diff
@@ -0,0 +1,172 @@
+		useStateExport := false
+		bufferedMaturityBlocks := v11.MaturityBlocks + v11.PendingBallotsDeletionBufferBlocks
+		currentHeight := int64(0)
+		if useStateExport {
+			currentHeight = 10309298
+		} else {
+			currentHeight = 144500
+		}
+		// Only stale ballots
+		startHeight := currentHeight - bufferedMaturityBlocks
+		endHeight := int64(0)
+
+		k, ctx, _, _ := keepertest.ObserverKeeper(t)
+		SaveBallotsToState(t, ctx, k, useStateExport, startHeight, endHeight)
+
+		ctx = ctx.WithBlockHeight(currentHeight)
+		err := v9MigrateStore(ctx, *k)
+		require.NoError(t, err)
+
+		require.Greater(t, len(k.GetAllBallots(ctx)), 0)
+
+		err = v11.MigrateStore(ctx, *k)
+		require.NoError(t, err)
+
+		deletionHeight := currentHeight - bufferedMaturityBlocks
+
+		ballots := k.GetAllBallots(ctx)
+		for _, ballot := range ballots {
+			require.GreaterOrEqual(t, ballot.BallotCreationHeight, deletionHeight)
+		}
+	})
+
+	t.Run("do not nothing if no ballots are present", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.ObserverKeeper(t)
+		ballotsBeforeMigrations := k.GetAllBallots(ctx)
+
+		err := v11.MigrateStore(ctx, *k)
+		require.NoError(t, err)
+
+		require.Len(t, k.GetAllBallots(ctx), len(ballotsBeforeMigrations))
+	})
+
+	t.Run("do not nothing if no stale ballots are present(Simulate Mainnet)", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.ObserverKeeper(t)
+		currentHeight := int64(144500)
+		bufferedMaturityBlocks := v11.MaturityBlocks + v11.PendingBallotsDeletionBufferBlocks
+
+		// No stale ballots ( All created ballots are higher than maturity blocks)
+		startHeight := currentHeight
+		endHeight := currentHeight - bufferedMaturityBlocks
+
+		SaveBallotsToState(t, ctx, k, false, startHeight, endHeight)
+		ballotsBeforeMigrations := k.GetAllBallots(ctx)
+
+		err := v11.MigrateStore(ctx, *k)
+		require.NoError(t, err)
+
+		require.Len(t, k.GetAllBallots(ctx), len(ballotsBeforeMigrations))
+
+		deletionHeight := currentHeight - bufferedMaturityBlocks
+		ballots := k.GetAllBallots(ctx)
+		for _, ballot := range ballots {
+			require.GreaterOrEqual(t, ballot.BallotCreationHeight, deletionHeight)
+		}
+	})
+
+}
+
+func v9MigrateStore(ctx sdk.Context, observerKeeper keeper.Keeper) error {
+	currentHeight := ctx.BlockHeight()
+	bufferedMaturityBlocks := v11.MaturityBlocks + v11.PendingBallotsDeletionBufferBlocks
+	// Maturity blocks is a parameter in the emissions module
+	if currentHeight < bufferedMaturityBlocks {
+		return nil
+	}
+	maturedHeight := currentHeight - bufferedMaturityBlocks
+	for i := maturedHeight; i > 0; i-- {
+		ballotList, found := observerKeeper.GetBallotListForHeight(ctx, i)
+		if !found {
+			continue
+		}
+		for _, ballotIndex := range ballotList.BallotsIndexList {
+			observerKeeper.DeleteBallot(ctx, ballotIndex)
+		}
+		observerKeeper.DeleteBallotListForHeight(ctx, i)
+	}
+	return nil
+}
+
+func ImportData(t *testing.T, cdc codec.JSONCodec) []*types.Ballot {
+	file := os.Getenv("STATE_EXPORT_PATH")
```

Comment:
Used file for local testing, the file size is 500 mb +

`zetacored export  --modules-to-export observer --output-document observer-state.json --log_format json`
can be used on a node to create the export